src/components/editor/TextEditor/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { ReactComponent as LeftAngleBraketIcon } from 'assets/svg/angle-braket.svg';
+import { ReactComponent as Picture } from 'assets/svg/picture.svg';
+import { ReactComponent as Plus } from 'assets/svg/plus.svg';
+import StarRating from 'components/StarRating';
+import Wysiwyg, { WysiwygRef } from 'components/editor/Wysiwyg';
+import { useNavigate } from 'react-router-dom';
+import { useEffect, useRef, useState } from 'react';
+import cn from 'utils/ts/classNames';
+import styles from './TextEditor.module.scss';
+
+interface TextEditorProps {
+  shop: string | undefined;
+}
+
+function TextEditor({ shop }: TextEditorProps) {
+  const [showTextTools, setShowTextTools] = useState(false);
+  const [isShopExist, setIsShopExist] = useState(false);
+  const [isRate, setIsRate] = useState(false);
+  const navigate = useNavigate();
+  const wysiwygRef = useRef<WysiwygRef>(null);
+
+  useEffect(() => {
+    if (shop === undefined) {
+      setIsShopExist(false);
+    } else {
+      setIsShopExist(true);
+    }
+  }, [shop]);
+
+  const textToolHandler = () => {
+    setShowTextTools(!showTextTools);
+  };
+  const getShop = () => {
+    // 여기서 Shop 정보를 가져온다.
+    // Shop 정보가 가져와졌다는 가정하에 true를 전달.
+    // 추가적으로 shop이름을 받아오는 로직 작성 필요.
+    setIsShopExist(true);
+  };
+  const rating = () => {
+    setIsRate(true);
+  };
+
+  return (
+    <div className={cn({
+      [styles.template]: true,
+      [styles['template--active']]: isShopExist,
+    })}
+    >
+      <div className={styles.header}>
+        <LeftAngleBraketIcon type="button" className={styles['header__button--prev']} onClick={() => navigate(-1)} />
+        { !isShopExist && <Plus type="button" className={styles['header__button--add']} onClick={getShop} /> }
+        { isShopExist && <div className={styles.header__shopname}>{ shop }</div> }
+        { isShopExist && <StarRating rating={rating} /> }
```

Comment:
앞선 상태를 제거하고 다음과 같이 수정할 수 있어 보입니다.

```suggestion
        { shop == null ? 
          <Plus 
              type="button" 
              className={styles['header__button--add']} 
              onClick={getShop} 
            />
          : (
          <> 
            <div className={styles.header__shopname}>{ shop }</div>
            <StarRating rating={rating} />
          </>
          )
        }
```

Reply:
아 이렇게 하면 되는군요..! 바로 이해가 되었습니다! 감사합니다!

Reply:
넵 ㅎㅎ react 상태의 Single Source of Truth 를 지키도록 노력해주세요 😄 https://reactjs.org/docs/lifting-state-up.html#lessons-learned